controller/Publish.php

```diff
@@ -55,15 +58,26 @@ public function index(){
 
     public function publish(){
         try {
-            $testUri = $this->getRequestParameter('testUri');
-            $classUri = $this->getRequestParameter('classUri');
+            $parsedBody = $this->getPsrRequest()->getParsedBody();
+            $testUri = $parsedBody['testUri'] ?? null;
+            $classUri = $parsedBody['classUri'] ?? null;
             $test = $this->getResource($testUri);
+
+            /** @var taoTests_models_classes_TestsService $testService */
+            $testService = $this->getServiceLocator()->get(taoTests_models_classes_TestsService::class);
+            $testItems = $testService->getTestItems($test);
+            if (empty($testItems)) {
+                throw new RuntimeException(
```

Comment:
With a little refactoring, it seems that only three things can throw an exception in the whole try:
- the two calls to getServiceLocator()->get(), which would mean a wiring problem and should crash the application
- the exception thrown here manually.
=> we can directly return the failure response.